modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostGISDialect.java

```diff
@@ -1407,14 +1409,26 @@ public void encodeGeometryValue(Geometry value, int dimension, int srid, StringB
 
     @Override
     public FilterToSQL createFilterToSQL() {
-        PostgisFilterToSQL sql = new PostgisFilterToSQL(this);
+        PostgisFilterToSQL sql =
+                new PostgisFilterToSQL(this, postgresMajorVersionIsHigherThen(PGSQL_V_12_0));
         sql.setLooseBBOXEnabled(looseBBOXEnabled);
         sql.setEncodeBBOXFilterAsEnvelope(encodeBBOXFilterAsEnvelope);
         sql.setFunctionEncodingEnabled(functionEncodingEnabled);
         sql.setEscapeBackslash(escapeBackslash);
         return sql;
     }
 
+    public boolean postgresMajorVersionIsHigherThen(Version version) {
```

Comment:
it actually returns the value from a equal or greater than comparison... 

```suggestion
    public boolean postgresMajorVersionIsEqualOrGreaterThan(Version version) {
```
Aside from that not sure if this needs to be `public`; I would think it is only used in the PG module

Reply:
Renamed and moved to FilterToSqlHelper (since it is used only there).